esx_service/utils/auth_data.py

```diff
@@ -572,30 +648,31 @@ def _remove_volumes_for_tenant(self, tenant_id):
             logging.error("Error %s when querying from tenants table", e)
             return str(e)
 
-        logging.debug("remove_volumes_for_tenant: %s %s", tenant_id, result)
-        tenant_name = result[0]
-        vmdks = vmdk_utils.get_volumes(tenant_name)
-        # Delete all volumes for this tenant. 
-        dir_path = None
-        for vmdk in vmdks:
-            vmdk_path = os.path.join(vmdk['path'], "{0}".format(vmdk['filename']))
-            if not dir_path:
-                dir_path = vmdk['path']
-            print("path=%s filename=%s", vmdk['path'], vmdk['filename'])    
-            logging.debug("Deleting volume path%s", vmdk_path)
-            err = vmdk_ops.removeVMDK(vmdk_path)
-            if err:
-                logging.error("remove vmdk %s failed with error %s", vmdk_path, err)
+        if result:    
+            logging.debug("remove_volumes_for_tenant: %s %s", tenant_id, result)
+            tenant_name = result[0]
+            vmdks = vmdk_utils.get_volumes(tenant_name)
+            # Delete all volumes for this tenant. 
+            dir_path = None
+            for vmdk in vmdks:
+                vmdk_path = os.path.join(vmdk['path'], "{0}".format(vmdk['filename']))
+                if not dir_path:
+                    dir_path = vmdk['path']
+                logging.debug("path=%s filename=%s", vmdk['path'], vmdk['filename'])    
+                logging.debug("Deleting volume path%s", vmdk_path)
+                err = vmdk_ops.removeVMDK(vmdk_path)
                 if err:
-                    return err
-        
-        # Delete path /vmfs/volumes/datastore_name/tenant_name
-        logging.debug("Deleting dir path%s", dir_path)
-        try:
-            os.rmdir(dir_path)
-        except os.error as e:
-            logging.error("remove dir %s failed with error %s", dir_path, e)
-            return str(e)
+                    logging.error("remove vmdk %s failed with error %s", vmdk_path, err)
+                    if err:
+                        return err
```

Comment:
Instead, remove as many of the volumes but keep track of any remove errors and at the end of the loop figure if there was an error in removing any of the volumes and then return error after logging.